realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -216,10 +233,9 @@ private void emitAccessors(JavaWriter writer) throws IOException {
                 writer.emitAnnotation("Override");
                 writer.beginMethod(fieldTypeCanonicalName, metadata.getGetter(fieldName), EnumSet.of(Modifier.PUBLIC));
                 writer.beginControlFlow("if (row.isNullLink(%s))", staticFieldIndexVarName(field));
-                writer.emitStatement("return null");
-                writer.endControlFlow();
-                writer.emitStatement(
-                        "return realm.get(%s.class, row.getLink(%s))",
+                        writer.emitStatement("return null");
```

Comment:
Yes, this is to mimic the code being generated, which makes what is being generated a bit more visible.